test/e2e/multihoming.go

```diff
@@ -629,12 +630,46 @@ var _ = Describe("Multi Homing", func() {
 
 		table.DescribeTable(
 			"multi-network policies configure traffic allow lists",
-			func(netConfig networkAttachmentConfig, allowedClientPodConfig podConfiguration, blockedClientPodConfig podConfiguration, serverPodConfig podConfiguration, policy *mnpapi.MultiNetworkPolicy) {
+			func(netConfig networkAttachmentConfig, allowedClientPodConfig podConfiguration, blockedClientPodConfig podConfiguration, serverPodConfig podConfiguration, policy *mnpapi.MultiNetworkPolicy, extraNamespaces ...v1.Namespace) {
+				for i := range extraNamespaces {
+					createdNamespace, err := cs.CoreV1().Namespaces().Create(context.Background(), &extraNamespaces[i], metav1.CreateOptions{})
+					Expect(err).NotTo(HaveOccurred())
+					extraNamespaces[i] = *createdNamespace
+					netConfig.namespace = createdNamespace.Name
+					allowedClientPodConfig.namespace = createdNamespace.Name
+					By("creating the attachment configuration for the extra-namespace")
+					_, err = nadClient.NetworkAttachmentDefinitions(createdNamespace.Name).Create(
+						context.Background(),
+						generateNAD(netConfig),
+						metav1.CreateOptions{},
+					)
+					Expect(err).NotTo(HaveOccurred())
+				}
+
+				blockUntilEverythingIsGone := metav1.DeletePropagationForeground
+				defer func() {
+					for _, ns := range extraNamespaces {
+						Expect(cs.CoreV1().Namespaces().Delete(
+							context.Background(),
+							ns.Name,
+							metav1.DeleteOptions{PropagationPolicy: &blockUntilEverythingIsGone},
+						)).To(Succeed())
+						Eventually(func() bool {
+							_, err := cs.CoreV1().Namespaces().Get(context.Background(), ns.Name, metav1.GetOptions{})
+							nsPods, podCatchErr := cs.CoreV1().Pods(ns.Name).List(context.Background(), metav1.ListOptions{})
+							return podCatchErr == nil && errors.IsNotFound(err) && len(nsPods.Items) == 0
+						}, 2*time.Minute, 5*time.Second).Should(BeTrue())
+					}
+				}()
```

Comment:
I think I'll create a separate context so I can create these extra things in the test setup / teardown. 

Cleaning up the side effects in a defer in a test is not good.